packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -0,0 +1,269 @@
+   */
+  readonly prefixPath: string;
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute
+ */
+export interface GrpcRouteMatch {
+  /**
+   * The fully qualified domain name for the service to match from the request
+   */
+  readonly serviceName: string;
+}
+
+/**
+ * Properties specific for HTTP Based Routes
+ */
+export interface HttpRouteSpecOptions {
+  /**
+   * The criterion for determining a request match for this Route
+   *
+   * @default - matches on '/'
+   */
+  readonly match?: HttpRouteMatch;
+
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  readonly weightedTargets: WeightedTarget[];
+}
+
+/**
+ * Properties specific for a TCP Based Routes
+ */
+export interface TcpRouteSpecOptions {
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  readonly weightedTargets: WeightedTarget[];
+}
+
+/**
+ * Properties specific for a GRPC Based Routes
+ */
+export interface GrpcRouteSpecOptions {
+  /**
+   * The criterion for determining a request match for this Route
+   *
+   * @default - no default
+   */
+  readonly match: GrpcRouteMatch;
+
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  readonly weightedTargets: WeightedTarget[];
+}
+
+/**
+ * All Properties for GatewayRoute Specs
+ */
+export interface RouteSpecConfig {
+  /**
+   * The spec for an http route
+   *
+   * @default - no http spec
+   */
+  readonly httpRouteSpec?: CfnRoute.HttpRouteProperty;
+
+  /**
+   * The spec for an http2 route
+   *
+   * @default - no http2 spec
+   */
+  readonly http2RouteSpec?: CfnRoute.HttpRouteProperty;
+
+  /**
+   * The spec for a grpc route
+   *
+   * @default - no grpc spec
+   */
+  readonly grpcRouteSpec?: CfnRoute.GrpcRouteProperty;
+
+  /**
+   * The spec for a tcp route
+   *
+   * @default - no tcp spec
+   */
+  readonly tcpRouteSpec?: CfnRoute.TcpRouteProperty;
+}
+
+/**
+ * Used to generate specs with different protocols for a RouteSpec
+ */
+export abstract class RouteSpec {
+  /**
+   * Creates an HTTP Based RouteSpec
+   */
+  public static http(options: HttpRouteSpecOptions): RouteSpec {
+    return new HttpRouteSpec(options, Protocol.HTTP);
+  }
+
+  /**
+   * Creates an HTTP2 Based RouteSpec
+   *
+   */
+  public static http2(options: HttpRouteSpecOptions): RouteSpec {
+    return new HttpRouteSpec(options, Protocol.HTTP2);
+  }
+
+  /**
+   * Creates a TCP Based RouteSpec
+   */
+  public static tcp(options: TcpRouteSpecOptions): RouteSpec {
+    return new TcpRouteSpec(options);
+  }
+
+  /**
+   * Creates a GRPC Based RouteSpec
+   */
+  public static grpc(options: GrpcRouteSpecOptions): RouteSpec {
+    return new GrpcRouteSpec(options);
+  }
+
+  /**
+   * List of targets that traffic is routed to when a request matches the route
+   */
+  abstract readonly weightedTargets: WeightedTarget[];
```

Comment:
Let's get rid of this property. The public API of this class should be the minimum we need.